### ISSUE TITLE
Consider using hostname to match ansible-pull hosts.

### ISSUE THREAD

**Issue by jody-frankowski** [ansible/ansible#11117]
**Issue Type:**
Bug report

Currently (according to https://github.com/ansible/ansible/blob/fe6a2d644dc610d713f96e9cef663b81c9762687/bin/ansible-pull#L165 ), ansible-pull tries to match hostnames against "localhost" "127.0.0.1" and the machine fqdn.

There is two problems with this approach:
- Many machines don't have an fqdn properly configured.
- Not all people want to use fqdns in inventory lists.

If there is no possible breakage, I think a simple fix would be to include the simple hostname in that matching list.

**Comment by ghost** [comment]
:+1: Like this idea a lot.

**Comment by bcoca** [comment]
something like this?

```
diff --git a/lib/ansible/cli/pull.py b/lib/ansible/cli/pull.py
index 620ec68..0555d1a 100644
--- a/lib/ansible/cli/pull.py
+++ b/lib/ansible/cli/pull.py
@@ -18,6 +18,7 @@
 ########################################################
 import datetime
 import os
+import platform
 import random
 import shutil
 import socket
@@ -107,7 +108,9 @@ class PullCLI(CLI):

         # Build Checkout command
         # Now construct the ansible command
-        limit_opts = 'localhost:%s:127.0.0.1' % socket.getfqdn()
+        node = platform.node()
+        host = socket.getfqdn()
+        limit_opts = 'localhost:%s:127.0.0.1' % ':'.join(set([host, node, host.split('.')[0], node.split('.')[0]]))
         base_opts = '-c local "%s"' % limit_opts
         if self.options.verbosity > 0:
             base_opts += ' -%s' % ''.join([ "v" for x in range(0, self.options.verbosity) ])
lines 1-48/48 (END)
```

**Comment by bcoca** [comment]
fixed via 9f29e39deaaf4a674d6730502210fbaf6c057c60

**Comment by jody-frankowski** [comment]
Looks good to me. Awesome, thanks.
